plugins/completion/codeium-nvim.nix

```diff
@@ -0,0 +1,116 @@
+{
+  lib,
+  ...
+}:
+let
+  inherit (lib.nixvim) defaultNullOpts;
+in
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "codeium-nvim";
+  originalName = "codeium.nvim";
+  luaName = "codeium";
+
+  maintainers = with lib.maintainers; [
+    GaetanLepage
+    khaneliman
+  ];
+
+  # TODO: added 2024-09-03 remove after 24.11
+  deprecateExtraOptions = true;
+  optionsRenamedToSettings = [
+    "configPath"
+    "binPath"
+    [
+      "api"
+      "host"
+    ]
+    [
+      "api"
+      "port"
+    ]
+    [
+      "tools"
+      "uname"
+    ]
+    [
+      "tools"
+      "uuidgen"
+    ]
+    [
+      "tools"
+      "curl"
+    ]
+    [
+      "tools"
+      "gzip"
+    ]
+    [
+      "tools"
+      "languageServer"
+    ]
+    "wrapper"
+  ];
+
+  imports = [
+    { cmpSourcePlugins.codeium = "codeium-nvim"; }
+  ];
+
+  settingsOptions = {
+    config_path = defaultNullOpts.mkStr {
+      __raw = "vim.fn.stdpath('cache') .. '/codeium/config.json'";
+    } "The path to the config file, used to store the API key.";
+
+    bin_path = defaultNullOpts.mkStr {
+      __raw = "vim.fn.stdpath('cache') .. '/codeium/bin'";
+    } "The path to the directory where the Codeium server will be downloaded to.";
+
+    api = {
+      host = defaultNullOpts.mkStr "server.codeium.com" ''
+        The hostname of the API server to use.
+      '';
+
+      port = defaultNullOpts.mkPositiveInt 443 ''
+        The port of the API server to use.
+      '';
```

Comment:
The old option was (effectively) `coercedTo ints.positive toString str`.

https://github.com/nix-community/nixvim/blob/17f4aa555615df37ec68ed75a04b8a8b1d36fd98/plugins/completion/cmp/sources/codeium-nvim.nix#L66

1. Does the plugin support lua numbers as ports, or does it expect numeric strings?
2. If end-users had string definitions in their old config, these will now fail the type-check.

Reply:
By looking at their config example and the usage, I would think just a string? 

https://github.com/Exafunction/codeium.nvim/blob/f6a2ef32a9e923cb0104a19d3e426b0e40e49505/lua/codeium/api.lua#L167-L173

Should we limit it to just strs or keep handling the number?

Reply:
Ideally we should match the upstream type, in this case it'd be `types.strMatching "[0-9]+"`, I think?

Because we have the rename, we need to still _support_ ints, but we can do this using `transitionType`, right?